retrofit/src/main/java/retrofit/RetrofitError.java

```diff
@@ -117,14 +123,14 @@ public Object getBodyAs(Type type) {
     if (response == null) {
       return null;
     }
-    TypedInput body = response.getBody();
+    ResponseBody body = response.body();
     if (body == null) {
       return null;
     }
     try {
       return converter.fromBody(body, type);
     } catch (IOException e) {
-      throw new RuntimeException(e); // Body is a byte[], can't be a real IO exception.
+      throw new RuntimeException(e); // Body is a Buffer, can't be a real IO exception.
```

Comment:
That's malformed at the HTTP transport layer, and it would have thrown by here (we consumed the entire body into a `Buffer`).